### DISCUSSION TITLE
Error (Xcode): Signing for "Runner" requires a development team

### DISCUSSION THREAD

**Discussion by mabehiry** [codemagic-ci-cd/codemagic-docs#2507] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

    scripts:
      - name: Set up code signing settings on Xcode project
        script: |
          xcode-project use-profiles

      - name: Get Flutter packages
        script: |
          flutter packages pub get

      - name: Create IOS
        script: |
          flutter create . --platforms ios

      - name: Install pods
        script: |
          find . -name "Podfile" -execdir pod install \;

      - name: Flutter build ipa
        script: |
          flutter build ipa --release \
            --build-name=1.0.0 \
            --build-number=$(($(app-store-connect get-latest-app-store-build-number "$APP_STORE_APPLE_ID") + 1)) \
            --export-options-plist=/Users/builder/export_options.plist

### Expected results

To build the IOS

### Actual results

Found build number 10 from App Store version 10

Archiving **.**.clone...
Developer identity "Apple Development: Created via API (**)" selected for iOS code signing
Running pod install...                                           1,150ms
Running Xcode build...                                          
Xcode archive done.                                          4.0s
Failed to build iOS app
Error (Xcode): Signing for "Runner" requires a development team. Select a development team in the Signing & Capabilities editor.
/Users/builder/clone/ios/Runner.xcodeproj

Encountered error while archiving for device.

Build failed :|
Step 8 script `Flutter build ipa` exited with status code 1

### Build id (optional)

6637d309eeccae0a818d5702

**Comment by dtrdic** [comment, marked as answer]
Hello @mabehiry since you are distributing your app to App Store i advise you to use a Distribution certificate and distribution provisioning profile, not development.
I can see you have uploaded provisioning profiles to Codemagic but their bundle ID says (*), please create a provisioning profile with the specific bundle Identifier and reference it in your yaml under the environment section.
```
    environment:
      ios_signing:
        distribution_type: app_store
        bundle_identifier: com.example.test
```
Also make sure that in your `project.pbxproj `file, the application `PRODUCT_BUNDLE_IDENTIFIER` is the same bundle as the one you have defined in your provisioning profile

**Reply by mabehiry** [reply]
It worked. Thank you!
